framework/db/oci/QueryBuilder.php

```diff
@@ -160,4 +160,61 @@ public function addForeignKey($name, $table, $columns, $refTable, $refColumns, $
 
         return $sql;
     }
+
+    /**
+     * Generates a batch INSERT SQL statement.
+     * For example,
+     *
+     * ~~~
+     * $sql = $queryBuilder->batchInsert('user', ['name', 'age'], [
+     *     ['Tom', 30],
+     *     ['Jane', 20],
+     *     ['Linda', 25],
+     * ]);
+     * ~~~
+     *
+     * Note that the values in each row must match the corresponding column names.
+     *
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column names
+     * @param array $rows the rows to be batch inserted into the table
+     * @return string the batch INSERT SQL statement
+     */
+    public function batchInsert($table, $columns, $rows)
+    {
+        $schema = $this->db->getSchema();
+        if (($tableSchema = $schema->getTableSchema($table)) !== null) {
+            $columnSchemas = $tableSchema->columns;
+        } else {
+            $columnSchemas = [];
+        }
+
+        $values = [];
+        foreach ($rows as $row) {
+            $vs = [];
```

Comment:
This is copied from the parent class, not my own code. If this to be refactored, it should be changed in other schemas, like sqlite.